### DISCUSSION TITLE
LS won't write a log file - permission denied?

### DISCUSSION THREAD

**Discussion by Radio-biscuit** [savonet/liquidsoap#2976] [Q&A]
Hello there, I'm very new to LS and rather green.

I had LS working with a trial script and was getting along nicely with it (lots of questions and confusion). I shut down my (Windows) machine at the end of the day and the started everything up again after a good night's sleep, only to find that the script I have been working on fails now with:

Logs can be found in "C:\\Users\\BLAH\AppData\\Local\\Temp\\liquidsoap-600-103aa8.log".

[wav @ 00000222a8da4900] Discarding ID3 tags because more suitable tags were found.
[wav @ 00000222a8da4900] Discarding ID3 tags because more suitable tags were found.
init: exception encountered during stop phase:
  Sys_error("C:\\Users\\BLAH\AppData\\Local\\Temp\\liquidsoap-600-103aa8.log: Permission denied")

This is where it was logging before I shut down the PC, but now, er, what's happened? Maybe I typed too hard and hurt its feelings?

If anyone knows how this happens, I'd be really grateful for a solution.

Thanks!

**Comment by Radio-biscuit** [comment]
Ooops, it's not the log file that gets permission denied, it's the error being written into the log file that's telling me of a permission denied problem on the output device. Sorry!

How I have to work on how to determine the right device to output to.

**Reply by toots** [reply]
Sounds good. For windows, I believe you can use `--list-portaudio-devices` to list all available output devices. All the best with your project!

**Reply by Radio-biscuit** [reply]
Thanks. LOL. I have 78 devices across 4 APIs. Now I have to work out which API is which and why some have more devices than others and how to select the same one each time after any changes. Ho hum .... :)
